samples/bluetooth/mesh/sensor_client/src/model_handler.c

```diff
@@ -58,11 +82,25 @@ static struct k_work_delayable get_data_work;
 
 static void get_data(struct k_work *work)
 {
-	static bool req_temp_range;
+	if (!bt_mesh_is_provisioned()) {
+		k_work_schedule(&get_data_work, K_MSEC(GET_DATA_INTERVAL));
+		return;
+	}
+
+	static uint32_t sensor_idx;
 	int err;
 
 	/* Only one message can be published at a time. Swap sensor after each timeout. */
-	if (req_temp_range) {
+	switch ((sensor_idx++ % 3)) {
+	case (0): {
```

Comment:
Why the parentheses?

Reply:
Operator precedence is scary! Removed them now.